projects/sandbox/train/train/utils.py

```diff
@@ -35,113 +35,122 @@ def split(X: Tensor, frac: float, axis: int) -> Tuple[Tensor, Tensor]:
         return torch.split(X, splits, dim=axis)
 
 
-def prepare_augmentation(
-    glitch_dataset: Path,
-    waveform_dataset: Path,
-    ifos: List[str],
-    train_start: float,
-    train_stop: float,
-    glitch_prob: float,
-    waveform_prob: float,
-    glitch_downweight: float,
-    swap_frac: float,
-    mute_frac: float,
-    sample_rate: float,
-    highpass: float,
-    max_min_snr: float,
-    min_min_snr: float,
-    max_snr: float,
-    snr_alpha: float,
-    snr_decay_steps: int,
-    invert_prob: float = 0.5,
-    reverse_prob: float = 0.5,
-    trigger_distance: float = 0,
-    valid_frac: Optional[float] = None,
-):
-    # build a glitch sampler from a pre-saved bank of
-    # glitches which will randomly insert them into
-    # either or both interferometer channels
-    glitch_dict = {}
-    valid_glitches = []
+def get_background(background_dir: Path):
+    # load in the data from the first segment in the background directory
+    try:
+        background_dataset = next(background_dir.iterdir())
+    except StopIteration:
+        raise ValueError(
+            f"No files in background data directory {background_dir}"
+        )
+    background = []
+
+    with h5py.File(background_dataset, "r") as f:
+        ifos = list(f.keys())
+        for ifo in ifos:
+            hoft = f[ifo][:]
+            background.append(hoft)
+        t0 = f[ifo].attrs["x0"]
+    return np.stack(background), ifos, t0
 
-    # calculate the time at which the validation set starts
-    full_duration = train_stop - train_start
-    if valid_frac is not None:
-        train_duration = (1 - valid_frac) * full_duration
-        valid_start = train_start + train_duration
-    else:
-        train_duration = full_duration
 
+def get_glitches(
+    glitch_dataset: Path, ifos: List[str], end_time: Optional[float] = None
+) -> GlitchSampler:
```

Comment:
This doesn't actually return a `GlitchSampler`

Reply:
Good catch, fixed